### DISCUSSION TITLE
Add a webpage builder interface

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#13735] [Feature Requests]
I imagine this to be waaay in the future, but how about a webpage builder for public sites. This could be used to build front-facing sites for the public and the admin-side could be used to create the layouts.

Something like [grapejs](https://github.com/artf/grapesjs) could be merged into the admin side to create templates that are then populated with data for the front-facing site. 

Dataflows could be used to create workflows for users who interact with the site as well. e.g. a user logs in and the workflow detects this user to be a good customer with things loaded in the cart, so the workflow shows a banner reminding them of the items in their cart to encourage checkout etc.

**Comment by emrahnazif** [comment]
I believe [Many-to-Any](https://docs.directus.io/configuration/relationships/#many-to-any-m2a) and repeater fields may work good to achieve your goal. Meanwhile, I do not think that grapesjs is suitable with directus. Shopify's theme editor style looks more suitable and it is almost as similar as repeaters except front-end view which is again not suitable with directus (I believe...). Maybe with some custom extension this can be also possible. 

And for your second example: 

> Dataflows could be used to create workflows for users who interact with the site as well. e.g. a user logs in and the workflow detects this user to be a good customer with things loaded in the cart, so the workflow shows a banner reminding them of the items in their cart to encourage checkout etc.

I also believe this is not job of flows. You can simply write a condition based on the json data you get to display anything to the visitor. 
Flow may be more useful to send them email if they forgot items in their cart etc.

**Comment by azrikahar** [comment]
There is an existing discussion for this in #5776, with GrapesJS being one of the suggestion! There's also a community extension using Editor.js instead: https://github.com/dimitrov-adrian/directus-extension-editorjs-interface

**Comment by rijkvanzanten** [comment]
The M2A Builder and (about to be released) EditorJS interface are both great webpage builders 🙂
